modules/atomic-widgets/elements/atomic-svg/atomic-svg.php

```diff
@@ -31,82 +30,73 @@ public function get_icon() {
 
 	protected function render() {
 		$settings = $this->get_atomic_settings();
+		$svg_url = $settings['svg']['src'] ?? '';
 
-		if ( ! isset( $settings['svg'] ) ) {
+		if ( ! $svg_url ) {
+			// we need this line in order to render the svg, otherwise it will be rendered as a string
+			printf( '%s', self::get_default_svg() ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 			return;
 		}
 
-		$svg_data = $settings['svg'];
-
-		$svg_content = isset( $svg_data['url'] ) ? $svg_data['url'] : '';
-
-		if ( strpos( $svg_content, self::SVG_ENCODING ) === 0 ) {
-			$svg_content = urldecode( substr( $svg_content, self::SVG_START_INDEX ) );
-		}
+		$svg_content = ( strpos( $svg_url, self::SVG_PREFIX ) === 0 )
+			? urldecode( substr( $svg_url, strlen( self::SVG_PREFIX ) ) )
```

Comment:
If you really want, you can create a union which sometimes be a URL and sometimes be a a real svg, then you can have 2 transformers for each prop type of the union